eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java

```diff
@@ -122,58 +112,27 @@ private Collection<String> discover(final Path file)
     }
 
     /**
-     * Xpath for selecting objects from given xml.
-     * @param versioned Select with versions or not.
-     * @return Xpath as list of strings
-     * @todo #1602:30min Simplify xpath. Current implementation for building
-     *  xpath with and without versions is quite ugly. For some reason
-     *  if we try to take `/concat(@base,'|',@ver)` and there are object without
-     *  attribute `ver` - xpath returns nothing. So we need to take `/@base`
-     *  from objects where attribute `ver` is not present in both cases and
-     *  then if flag `withVersions` is `true` - take `concat(@base,'|',@ver)`
-     *  from objects attribute `ver` is present.
-     */
-    private String xpath(final boolean versioned) {
-        final Collection<String> xpath = new ListOf<>(
-            "//o[",
-            "not(starts-with(@base,'.'))",
-            "and @base != 'Q'",
-            "and @base != '^'",
-            "and @base != '$'",
-            "and @base != '&'",
-            "and not(@ref)"
-        );
-        final List<String> tail;
-        if (versioned) {
-            tail = new ListOf<>(
-                "and @ver",
-                "]/concat(@base,'|',@ver)"
-            );
-        } else {
-            tail = new ListOf<>();
-            if (this.withVersions) {
-                tail.add("and not(@ver)");
-            }
-            tail.add("]/@base");
-        }
-        xpath.addAll(tail);
-        return String.join(
-            " ",
-            xpath
-        );
-    }
-
-    /**
-     * Get unique list of object names from given XML by provided xpath.
+     * Get unique list of object names from given XML.
      * @param xml XML.
-     * @param xpath Xpath.
-     * @return Iterable of object names.
+     * @return Object names.
      */
-    private static Set<String> names(final XML xml, final String xpath) {
+    private static Set<String> names(final XML xml) {
```

Comment:
@volodya-lombrozo suggestion: since this function is called only once, don't you think it can be removed and its content can be inlined in `discover` function?

Reply:
@maxonfjvipon I don't like this idea. I prefer using many small methods, where each performs a small unit of work. In this case, it's much easier to understand the code. Originally it isn't my idea:

In [Clean Code: A Handbook of Agile Software Craftsmanship](https://www.amazon.com/dp/0132350882), Robert Martin says:
> The first rule of functions is that they should be small. The second rule of functions is that they should be smaller than that. Functions should not be 100 lines long. Functions should hardly ever be 20 lines long.

Reply:
@volodya-lombrozo fair enough